Platform/ApollolakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -627,6 +627,46 @@ PlatformCpuInit (
   AsmWriteMsr64 (EFI_MSR_POWER_MISC, Data);
 }
 
+/**
+  Update current boot Payload ID.
+
+**/
+VOID
+UpdatePayloadId (
+  VOID
+  )
+{
+  GEN_CFG_DATA       *GenericCfgData;
+  PLATFORM_CFG_DATA  *PlatCfgData;
+  UINT32              PayloadId;
+  UINT32              GpioLevel;
+
+  PayloadId = GetPayloadId ();
+  GenericCfgData = (GEN_CFG_DATA *)FindConfigDataByTag (CDATA_GEN_TAG);
+  if (GenericCfgData != NULL) {
+    if (GenericCfgData->PayloadId == AUTO_PAYLOAD_ID_SIGNATURE) {
+      PayloadId = 0;
```

Comment:
Looks PayloadId is got in line 644. 
In this case, if the value is auto, should it be 0 or keep the original value?

Reply:
this is the first time we set Payload ID value.  The original should be the same as 0.
But I think we can remove this to simplify the code.  Will change the code.

Reply:
Just revisited the code logic, I think it is better to set it to 0 to boot OsLoader if PayloadId is AUTO in CFGDATA.  This is to enforce we will boot to a good known PayloadId in case that the original value is not valid for some reason. This makes the behavior more deterministic.

Reply:
Should we update Line 644 to be PayloadId = 0; if this is the first time to set PayloadId?
It is a little confusing if we get PayloadId in Line 644, and then update to 0 in line 648.

Reply:
OK, updated the code.